packages/@aws-cdk/aws-codepipeline-actions/lib/bitbucket/source-action.ts

```diff
@@ -0,0 +1,107 @@
+import * as codepipeline from '@aws-cdk/aws-codepipeline';
+import * as iam from '@aws-cdk/aws-iam';
+import { Construct } from '@aws-cdk/core';
+import { Action } from '../action';
+import { sourceArtifactBounds } from '../common';
+
+/**
+ * Construction properties for {@link BitBucketSourceAction}.
+ *
+ * @experimental
+ */
+export interface BitBucketSourceActionProps extends codepipeline.CommonAwsActionProps {
+  /**
+   * The output artifact that this action produces.
+   * Can be used as input for further pipeline actions.
+   */
+  readonly output: codepipeline.Artifact;
+
+  /**
+   * The ARN of the CodeStar Connection created in the AWS console
+   * that has permissions to access this BitBucket repository.
+   *
+   * @example 'arn:aws:codestar-connections:us-east-1:123456789012:connection/12345678-abcd-12ab-34cdef5678gh'
+   */
+  readonly connection: string;
+
+  /**
+   * The owning user or organization of the repository.
+   *
+   * @example 'aws'
+   */
+  readonly owner: string;
+
+  /**
+   * The name of the repository.
+   *
+   * @example 'aws-cdk'
+   */
+  readonly repo: string;
```

Comment:
I don't know that we need to split `owner` and `repo` (is it even called "owner" in bit bucket's terminology). As a user I would rather just specify `repo: "aws/aws-cdk"). We can add validation that this is the format.

Reply:
The thing is, in all places you can provide this ([CodePipeline GitHub source](https://github.com/aws/aws-cdk/blob/a2a2f130ffec176a01991376679eeccf1027707e/packages/%40aws-cdk/aws-codepipeline-actions/lib/github/source-action.ts#L44-L52), [CodeBuild GitHub source](https://github.com/aws/aws-cdk/blob/a2a2f130ffec176a01991376679eeccf1027707e/packages/%40aws-cdk/aws-codebuild/lib/source.ts#L576-L588) and [CodeBuild BitBucket source](https://github.com/aws/aws-cdk/blob/a2a2f130ffec176a01991376679eeccf1027707e/packages/%40aws-cdk/aws-codebuild/lib/source.ts#L665-L677)), we take it in as an `owner` / `repo` pair - so I would rather keep it consistent across our libraries).

Reply:
Ok